src/chainstate/stacks/db/blocks.rs

```diff
@@ -1538,13 +1538,15 @@ impl StacksChainState {
         }
         
         // does this block match the burnchain state? skip if not
-        let (commit_burn, sortition_burn) = match StacksChainState::validate_anchored_block_burnchain(burn_tx, burn_header_hash, block, self.mainnet, self.chain_id)? {
-            Some((commit_burn, sortition_burn)) => (commit_burn, sortition_burn),
-            None => { 
-                let msg = format!("Invalid block {}: does not correspond to burn chain state", block.block_hash());
-                warn!("{}", &msg);
-
-                return Err(Error::InvalidStacksBlock(msg));
+        let (commit_burn, sortition_burn) = match block.header.is_genesis() {
```

Comment:
This isn't correct.  The `is_genesis()` check only determines whether or not the block has a parent (a more appropriate method name would be `is_first()`).  But, anyone can mine a "first" block at any point in time by setting its parent to `000...000`.  As such, it will always have a non-zero commit burn and sortition burn.

Reply:
The reason this isn't tripping any test errors right now is because we aren't yet testing the block rewards rigorously (this is blocked on the need to measure computing usage).

Reply:
Reverted with https://github.com/blockstack/blockstack-core/pull/1202/commits/09be511b08fd289e81853816531411af48a814dc